google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/JdbcResultSetMetaDataTest.java

```diff
@@ -0,0 +1,512 @@
+    }
+  }
+
+  private int getSqlType(Type type) {
+    if (type == Type.bool())
+      return Types.BOOLEAN;
+    if (type == Type.bytes())
+      return Types.BINARY;
+    if (type == Type.date())
+      return Types.DATE;
+    if (type == Type.float64())
+      return Types.DOUBLE;
+    if (type == Type.int64())
+      return Types.BIGINT;
+    if (type == Type.string())
+      return Types.NVARCHAR;
+    if (type == Type.timestamp())
+      return Types.TIMESTAMP;
+    if (type.getCode() == Code.ARRAY)
+      return Types.ARRAY;
+    return Types.OTHER;
+  }
+
+  @Test
+  public void getColumnTypeName() throws SQLException {
+    int index = 1;
+    for (TestColumn col : TEST_COLUMNS) {
+      assertEquals(col.type.getCode().name(), subject.getColumnTypeName(index));
+      index++;
+    }
+  }
+
+  @Test
+  public void testIsReadOnly() throws SQLException {
+    for (int i = 0; i < TEST_COLUMNS.size(); i++) {
+      assertFalse(subject.isReadOnly(i));
+    }
+  }
+
+  @Test
+  public void testIsWritable() throws SQLException {
+    for (int i = 0; i < TEST_COLUMNS.size(); i++) {
+      assertTrue(subject.isWritable(i));
+    }
+  }
+
+  @Test
+  public void testIsDefinitelyWritable() throws SQLException {
+    for (int i = 0; i < TEST_COLUMNS.size(); i++) {
+      assertFalse(subject.isDefinitelyWritable(i));
+    }
+  }
+
+  @Test
+  public void testGetColumnClassName() throws SQLException {
+    for (int i = 1; i <= TEST_COLUMNS.size(); i++) {
+      assertEquals(getTypeClassName(TEST_COLUMNS.get(i - 1).type), subject.getColumnClassName(i));
+    }
+  }
+
+  private String getTypeClassName(Type type) {
+    if (type == Type.bool())
+      return Boolean.class.getName();
+    if (type == Type.bytes())
+      return byte[].class.getName();
+    if (type == Type.date())
+      return Date.class.getName();
+    if (type == Type.float64())
+      return Double.class.getName();
+    if (type == Type.int64())
+      return Long.class.getName();
+    if (type == Type.string())
+      return String.class.getName();
+    if (type == Type.timestamp())
+      return Timestamp.class.getName();
+    if (type.getCode() == Code.ARRAY) {
+      if (type.getArrayElementType() == Type.bool())
+        return Boolean[].class.getName();
+      if (type.getArrayElementType() == Type.bytes())
+        return byte[][].class.getName();
+      if (type.getArrayElementType() == Type.date())
+        return Date[].class.getName();
+      if (type.getArrayElementType() == Type.float64())
+        return Double[].class.getName();
+      if (type.getArrayElementType() == Type.int64())
+        return Long[].class.getName();
+      if (type.getArrayElementType() == Type.string())
+        return String[].class.getName();
+      if (type.getArrayElementType() == Type.timestamp())
+        return Timestamp[].class.getName();
+    }
+    return null;
+  }
+
+  @Test
+  public void testToString() {
+    assertNotNull(subject.toString());
```

Comment:
let's also assert what the string value is

Reply:
done